### DISCUSSION TITLE
Exchangelib Message fields

### DISCUSSION THREAD

**Discussion by pngom** [ecederstrand/exchangelib#1039] [Q&A] [answered]
Hi - I am iterating over email messages to extract some fields. However, I am getting an error "Message object has no attribute 'name of field'". To simplify, I replicated the errors with a simple print. I also used hasattr() to check which attributes belong to the object. I copied the field names from 
[https://docs.microsoft.com/en-us/exchange/client-developer/web-service-reference/message-ex15websvcsotherref](url)

[exchangelib message attributes error.pdf]

![image](https://user-images.githubusercontent.com/59926154/150409225-65077e12-e838-441d-a191-45167507153d.png)
 
![image](https://user-images.githubusercontent.com/59926154/150408739-548d64d8-28a5-41ce-8281-c951049ce603.png)

![image](https://user-images.githubusercontent.com/59926154/150409323-02ffb47b-2d26-40ba-b327-8f9832c40e39.png)

![image](https://user-images.githubusercontent.com/59926154/150408858-5812516b-b97a-4f99-8260-e467d7816411.png)

**Comment by ecederstrand** [comment, marked as answer]
Those are not the correct field names for the `Message` class in exchangelib. You can see field names at https://ecederstrand.github.io/exchangelib/exchangelib/items/message.html#exchangelib.items.message.Message or just by printing `Message.FIELDS`.

**Reply by pngom** [reply]
Thanks ecederstrand! I appreciate the quick and valuable answer
